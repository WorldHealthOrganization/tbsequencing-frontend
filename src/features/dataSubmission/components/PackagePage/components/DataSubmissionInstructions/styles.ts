import { css } from '@emotion/react';
import appColors from '../../../../../../styles/colors';

export const container = css({
  paddingTop: 51,
  paddingLeft: 40,
  paddingRight: 24,
});

export const closeIconSx = {
  fill: appColors.secondary.whoBlue,
  fontSize: 21,
};

export const closeIconWrapper = css({
  display: 'flex',
  justifyContent: 'flex-end',
});

export const header = css({
  marginBottom: 10,
  marginTop: 67,
});

export const paragraph = css({
  marginTop: 24,
});

export const downloadBtn = css({
  marginTop: 24,
});

export const info = css({
  paddingRight: 12,
  height: '62.15%',
  overflowY: 'auto',
});
