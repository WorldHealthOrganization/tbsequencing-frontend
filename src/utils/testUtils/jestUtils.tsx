import React, { ReactElement } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { reducer } from '../../app/store';

const getWrapper = (
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
  } = {},
) => ({ children }: { children: ReactElement }) => (
  <Provider store={store}>{children}</Provider>
);

function render(
  ui: ReactElement,
  {
    // @ts-ignore
    preloadedState,
    store = configureStore({ reducer, preloadedState }),
    ...renderOptions
  } = {},
) {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const getTreeJSONWithRedux = (ui: ReactElement, {
  // @ts-ignore
  preloadedState,
  store = configureStore({ reducer, preloadedState }),
} = {}) => {
  const Wrapper = getWrapper(ui, { preloadedState, store });

  const tree = renderer
    .create(<Wrapper>{ui}</Wrapper>)
    .toJSON();

  return tree;
};

export const wrappedInRouter = (
  component: ReactElement,
) => <BrowserRouter>{component}</BrowserRouter>;

export * from '@testing-library/react';
export { render, getTreeJSONWithRedux };
