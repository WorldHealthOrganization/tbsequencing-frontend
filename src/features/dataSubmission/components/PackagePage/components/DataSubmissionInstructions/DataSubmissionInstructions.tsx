/** @jsxImportSource @emotion/react */
import React from 'react';
import { IconButton } from '@mui/material';
import { Close } from '@mui/icons-material';
import * as styles from './styles';
import PrimaryText from '../../../../../../components/typography/PrimaryText';
import { useDrawerApi } from '../../../../../drawer/hooks/useDrawerApi';
import H1 from '../../../../../../components/typography/H1';
import { AppButton } from '../../../../../../components/AppButton/AppButton';
import { TEMPLATE_ARCHIVE_NAME, templateUrl } from '../../../../../../utils/templateFiles';
import { downloadFileViaAnchor } from '../../../../../../utils/fileDownload';

const instructions: string[] = [
  `To provide your MIC values or binary resistant/susceptible calls, please download 
  and use the template below.`,
  `Data for binary resistant/susceptible calls and MIC values will be read
  from the excel sheet named "PDST" and "MIC", respectively.`,
  `If you provided sample identification labels from the INSDC BioSample 
  database, they will be automatically linked to the publicly 
  available sequencing data.`,
  `Otherwise, you will need to upload the sequencing data via the provided
  interface.`,
];

export const DataSubmissionInstructions = () => {
  const { closeDrawer } = useDrawerApi();

  const onDownloadTemplateClick = () => {
    downloadFileViaAnchor(templateUrl, TEMPLATE_ARCHIVE_NAME);
  };

  return (
    <div css={styles.container}>
      <div css={styles.closeIconWrapper}>
        <IconButton onClick={closeDrawer}><Close sx={styles.closeIconSx} /></IconButton>
      </div>
      <H1 style={styles.header}>Instructions</H1>
      <div css={styles.info}>
        {instructions.map((
          paragraph,
        ) => <PrimaryText key={paragraph} style={styles.paragraph}>{paragraph}</PrimaryText>)}
      </div>
      <AppButton
        style={styles.downloadBtn}
        onClick={onDownloadTemplateClick}
        startIconName="file_download_icon"
        variant="outlined"
        size="medium"
      >
        Download Template
      </AppButton>
    </div>
  );
};

export default DataSubmissionInstructions;
