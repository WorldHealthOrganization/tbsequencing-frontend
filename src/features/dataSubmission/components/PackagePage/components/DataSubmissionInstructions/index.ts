import DataSubmissionInstructions from './DataSubmissionInstructions';

export default DataSubmissionInstructions;
