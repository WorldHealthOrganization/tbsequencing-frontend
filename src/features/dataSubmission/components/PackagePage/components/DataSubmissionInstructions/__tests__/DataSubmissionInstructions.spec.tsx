import React from 'react';
import { getTreeJSONWithRedux, render } from '../../../../../../../utils/testUtils/jestUtils';
import DataSubmissionInstructions from '../DataSubmissionInstructions';

describe('DataSubmissionInstructions component test', () => {
  test('renders DataSubmissionInstructions component without error', () => {
    render(
      <DataSubmissionInstructions />,
    );
  });

  it('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      <DataSubmissionInstructions />,
    );

    expect(tree).toMatchSnapshot();
  });
});
