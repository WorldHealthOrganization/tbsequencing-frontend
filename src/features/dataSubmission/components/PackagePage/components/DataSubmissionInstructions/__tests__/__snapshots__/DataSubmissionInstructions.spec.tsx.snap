// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSubmissionInstructions component test Should render correctly 1`] = `
<div
  className="css-scjdyf-DataSubmissionInstructions"
>
  <div
    className="css-1f0od2t-DataSubmissionInstructions"
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-10d54gz-MuiSvgIcon-root"
        data-testid="CloseIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </button>
  </div>
  <p
    className="MuiTypography-root MuiTypography-body1 css-1thlxsv-MuiTypography-root-BaseText"
  >
    Instructions
  </p>
  <div
    className="css-6cm2mt-DataSubmissionInstructions"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-o6df3b-MuiTypography-root-BaseText"
    >
      To provide your MIC values or binary resistant/susceptible calls, please download 
  and use the template below.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-o6df3b-MuiTypography-root-BaseText"
    >
      Data for binary resistant/susceptible calls and MIC values will be read
  from the excel sheet named "PDST" and "MIC", respectively.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-o6df3b-MuiTypography-root-BaseText"
    >
      If you provided sample identification labels from the INSDC BioSample 
  database, they will be automatically linked to the publicly 
  available sequencing data.
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-o6df3b-MuiTypography-root-BaseText"
    >
      Otherwise, you will need to upload the sequencing data via the provided
  interface.
    </p>
  </div>
  <button
    className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-11ged7r-MuiButtonBase-root-MuiButton-root-AppButton"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
    >
      <span
        aria-hidden={true}
        className="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-kp9ftd-MuiIcon-root"
      >
        file_download_icon
      </span>
    </span>
    <p
      className="MuiTypography-root MuiTypography-body1 css-3cwtjx-MuiTypography-root-BaseText"
    >
      Download Template
    </p>
  </button>
</div>
`;
